ArduPlane/GCS_Mavlink.cpp

```diff
@@ -1009,12 +985,11 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_long_packet(const mavlink_command_l
             if (!check_latlng(packet.param5,packet.param6)) {
                 return MAV_RESULT_FAILED;
             }
-            Location new_home_loc {};
-            new_home_loc.lat = (int32_t)(packet.param5 * 1.0e7f);
-            new_home_loc.lng = (int32_t)(packet.param6 * 1.0e7f);
-            new_home_loc.alt = (int32_t)(packet.param7 * 100.0f);
-            plane.set_home(new_home_loc);
-            AP::ahrs().lock_home();
+            set_home(Location{
```

Comment:
i'm not keen on the loss of labels here

Reply:
Labels are back in the form assigning directly into the elements.

Turns out I was using a gcc extension here - clang++ didn't like it.

So this PR is a lot simpler now.